### PR TITLE
feat(TUP-20458)Implement the resource dependencies editor page for DI

### PR DESCRIPTION
**Pull request by jding-tlnd**
jobs/joblets
https://jira.talendforge.org/browse/TUP-20458

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/dialog/DependenciesResourceSelectionDialog.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    private IRepositoryNode result;
+
+    public DependenciesResourceSelectionDialog(final Shell parentShell) {
+        super(parentShell);
+    }
+
+    protected void configureShell(Shell shell) {
+        super.configureShell(shell);
+        shell.setText("Select a Route Resource");
```
Comment (cmeng-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/dialog/DependenciesResourceSelectionDialog.java]: It's better to use i18n

File: main/plugins/org.talend.designer.runprocess/src/main/java/org/talend/designer/runprocess/java/TalendJavaProjectManager.java [hunk 2]
```diff
@@ -543,4 +548,26 @@ private static void generatePom(Item item, int option, IProcessor processor) thr
                 item.getProperty());
     }
 
+    private static void SynchronizeResourceFile(Property property) throws Exception {
```
Comment (cmeng-talend) [inline main/plugins/org.talend.designer.runprocess/src/main/java/org/talend/designer/runprocess/java/TalendJavaProjectManager.java]: It's better to lowercase the first char

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/AbstractMultiPageTalendEditor.java [hunk 3]
```diff
@@ -1060,6 +1068,18 @@ protected void createPage2() {
         }
     }
 
+    // create Dependencies editor
+    protected void createPage3() {
+        dependenciesEditor = new JobDependenciesEditor(this, designerEditor.isReadOnly());
+        try {
+            int index = addPage(dependenciesEditor, getEditorInput());
+            setPageText(index, "Dependencies");
```
[the remark this answers is not in the document]
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/AbstractMultiPageTalendEditor.java]: Solved, thanks

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/dialog/DependenciesResourceSelectionDialog.java (hunk 1 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/dialog/DependenciesResourceSelectionDialog.java]: Solved, thanks

File: main/plugins/org.talend.designer.runprocess/src/main/java/org/talend/designer/runprocess/java/TalendJavaProjectManager.java (hunk 2 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.runprocess/src/main/java/org/talend/designer/runprocess/java/TalendJavaProjectManager.java]: Sorry wrong write, solved, thanks

**Pull request merged**